lib/src/rules/prefer_relative_imports.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc =
+    r"Prefer relative paths when importing libraries within your own package's lib directory.";
+
+const _details = r'''
+
+When referencing a library inside your package’s `lib` directory from another library in that same package, prefer use relative URI.
```

Comment:
The word "referencing" implies to me either `import`, `export`, `part` or `part of` directives. Should this be expanded to cover the other cases?